rosbag2_compression/test/rosbag2_compression/fake_decompressor.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "rcpputils/filesystem_helper.hpp"
+
+#include "fake_decompressor.hpp"
```

Comment:
same comment as in #218 we might keep the mock interfaces.

Reply:
I didn't get a response on this from the previous PR but reiterating here:

This is technically a fake implementation and not a mock. I'm concerned of potential confusion down the line if we decide to call it a mock instead.
I will be making mocks for these down the line tho.